cmake/platform/linux/wayland.cmake

```diff
@@ -1,6 +1,21 @@
-set(PLATFORM_REQUIRED_DEPS OpenGl EGL Waylandpp LibDRM Xkbcommon)
```

Comment:
either OpenGL or GLES is required. none or both are not allowed. can we do it like this?

if opengl
set(PLATFORM_REQUIRED_DEPS OpenGl EGL Waylandpp LibDRM Xkbcommon)
else if gles
set(PLATFORM_REQUIRED_DEPS OpenGLES EGL Waylandpp LibDRM Xkbcommon)
endif